src/lib.rs

```diff
@@ -1267,22 +1183,12 @@ impl EGraph {
                         .iter()
                         .map(|e| e.1.clone())
                         .collect();
-                    // Same as above, avoid turning termdag into a string if we have messages disabled for performance
-                    if self.messages_enabled() {
-                        log::info!("extracted variants:");
-                        let mut msg = String::default();
-                        msg += "(\n";
-                        assert!(!terms.is_empty());
-                        for expr in &terms {
-                            let str = termdag.to_string(expr);
-                            log::info!("   {str}");
-                            msg += &format!("   {str}\n");
-                        }
-                        msg += ")";
-                        self.print_msg(msg);
+                    if log_enabled!(Level::Info) {
```

Comment:
Under the setting LOG_LEVEL=INFO and `--mode no-messages`, the new implementation would still produce a big output, unlike the old behavior, is this right?

Reply:
The old behavior did turn off some logging with no messages, and now they are entirely independent. If there is any logging that seems to verbose, we should probably change the log level or remove it?